File: src/main/kotlin/com/sainnt/netshop/entity/Product_.java
package com.sainnt.netshop.entity;


import javax.annotation.Generated;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(Product.class)
public abstract class Product_ {
    public static volatile SingularAttribute<Product, Integer> id;
    public static volatile SingularAttribute<Product, String> name;
    public static volatile SingularAttribute<Product, Long> price;
    public static volatile SingularAttribute<Product, Cart> catalog;
    public static volatile SingularAttribute<Product, ProductDescription> description;
}

File: src/main/kotlin/com/sainnt/netshop/entity/Product.kt
package com.sainnt.netshop.entity

import javax.persistence.*


@Entity
@SequenceGenerator(name = "productSequence", sequenceName = "PRODUCT_SEQ", initialValue = 1, allocationSize = 1)
class Product(
    @Column(name = "product_name", nullable = false)
    var name: String,
    @Column(nullable = false)
    var price: Long,
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="catalog_id", nullable = false)
    var catalog: Catalog
) {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "productSequence")
    val id: Long? = null

    @OneToOne(fetch = FetchType.LAZY, mappedBy = "product")
    var productDescription: ProductDescription? = null

}